app/clients/lider_client.rb

```diff
@@ -0,0 +1,47 @@
+class LiderClient < ChromeClient
+  BASE_URL = 'https://www.lider.cl/supermercado/search?Ntt='
+
+  def products_by_query(query: nil)
+    ensuring_browser_closure do
+      go_to_products_page(query)
+
+      scraped_products = []
+      browser.search('.product-details').wait(:present, timeout: 5.0).map do |product|
+        scraped_products.push({
+                                brand: get_brand(product),
+                                offer_price: get_offer_price(product),
+                                normal_price: get_normal_price(product),
+                                measure: get_measure(product),
+                                name: get_name(product)
+                              })
+        puts scraped_products.last
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.